### ISSUE TITLE
Support for tags

### ISSUE THREAD

**Issue by panieldark** [MetaMask/snaps#1475]
Hey all, would you mind supporting tags (in addition to versions, which you support) for npm snap packages? i.e, `npm:@foo@testnet` would work. 

This would allow snap builders to specify stable versions per intended environment, a feat which isn't quite supported as is.

Fwiw, I cloned locally and found that, syntactically, the [NpmSnapIdStruct](https://github.com/MetaMask/snaps/blob/main/packages/snaps-utils/src/snaps.ts#L266-L287) could just do a conditional check on whether a tag is appended at the end, (as well as append to the tests [here](https://github.com/MetaMask/snaps/blob/main/packages/rpc-methods/src/restricted/caveats/snapIds.test.ts#L31)?), but I don't know if that would actually equate to the tag actually being equated for later downstream.

This would be a huge add!

**Comment by panieldark** [comment]
Found https://github.com/MetaMask/snaps/issues/1475!

**Comment by ritave** [comment]
Do you have any source code confirming that they don't work?

**Comment by panieldark** [comment]
Yes, but it's a private repository. To reproduce, where you specify `npm:@foo/bar`, you specify `npm:@foo/bar@valid-tag`.

To tag a version, you publish with `yarn npm publish --tag YOUR_TAG_HERE --access public`

**Comment by panieldark** [comment]
Any updates on this?
